### PR TITLE
Additions To Documentation

### PR DESCRIPTION
**Pull request by Arsenikr**
- split context standards to mandatory and recommended
- added some more best practices.

Please review and add insights

## Status
In Progress

### CODE REVIEW

**Comment by fvigo** [comment]
Merged to new sidebar to avoid conflicts

**Review by fvigo — COMMENTED** [review]
Thanks @Arsenikr ! 

I made a few changes in the sidebar structure (aligned to the latest changed we made), and moved the Generic IAM Commands document under Advanced, I don't think we need a specific section for it right now (we might add it later as we have more similar articles)

I think the IAM command should be completed with all the examples before merging this. If you don't think you can complete it quickly, I suggest to remove it from this PR, we can merge this one and readd that once it's ready.

**Comment by Arsenikr** [comment]
added more fixes and additional things. please review

File: docs/playbooks/playbooks-generic.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+---
+
+## Generic playbooks mapped by use case
+
+|  Generic Playbook   |  Use Cases | 
+|---|---|
+| [Get Original Email - Generic](https://github.com/demisto/content/blob/master/Packs/CommonPlaybooks/Playbooks/Get_Original_Email_-_Generic_README.md)  | Email Gateway |  
+| [Account Enrichment Generic](https://github.com/demisto/content/blob/master/Packs/CommonPlaybooks/Playbooks/playbook-Account_Enrichment_-_Generic_v2.1_README.md)| IAM | 
+|[Block Account](https://github.com/demisto/content/blob/master/Packs/CommonPlaybooks/Playbooks/playbook-Account_Enrichment_-_Generic_v2.1_README.md)| IAM |
+| [Block File Generic](https://github.com/demisto/content/blob/master/Packs/CommonPlaybooks/Playbooks/playbook-Account_Enrichment_-_Generic_v2.1_README.md) | Endpoint |
```
Comment (fvigo) [inline docs/playbooks/playbooks-generic.md:13]: this link is not correct

File: docs/playbooks/playbooks-generic.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+title: Generic Playbooks
+---
+
+## Generic playbooks mapped by use case
+
+|  Generic Playbook   |  Use Cases | 
+|---|---|
+| [Get Original Email - Generic](https://github.com/demisto/content/blob/master/Packs/CommonPlaybooks/Playbooks/Get_Original_Email_-_Generic_README.md)  | Email Gateway |  
+| [Account Enrichment Generic](https://github.com/demisto/content/blob/master/Packs/CommonPlaybooks/Playbooks/playbook-Account_Enrichment_-_Generic_v2.1_README.md)| IAM | 
+|[Block Account](https://github.com/demisto/content/blob/master/Packs/CommonPlaybooks/Playbooks/playbook-Account_Enrichment_-_Generic_v2.1_README.md)| IAM |
```
Comment (fvigo) [inline docs/playbooks/playbooks-generic.md:12]: this link is not correct

**Review by fvigo — CHANGES_REQUESTED** [review]
partial review, found some broken links

File: docs/integrations/generic-commands-about.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+id: generic-commands-about
+title: Generic Commands
+---
+
+
+## About Generic Commands
+
+Today the current XSOAR platform supports over 450+ integrations. 
+Some commands can be generalized across similar integrations. It allows the  
+These commands can be used both on all integrations or with `using` parameter on a specific instance of an integration.
```
Comment (fvigo) [inline docs/integrations/generic-commands-about.md:11]: Can you be a little more specific in describing what these commands are and why are important, maybe with some examples?

Also is `using` documented somewhere in the product doc you can link to?

File: docs/integrations/context-standards-recommended.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+## Ticket
+The following is the format for a ticket.
+```json
+"Ticket": {
+    "ID": "STRING, The ID of the ticket.",
+    "Creator": "STRING, The user who created the ticket.",
+    "Assignee": "STRING, The user assigned to the ticket.",
+    "State": "STRING, The status of the ticket. Can be "closed", "open", or "on hold"."
+}
```
Comment (David-BMS) [inline docs/integrations/context-standards-recommended.md:16]: How about a ticket summary or description?

File: docs/concepts/use-cases.md [hunk 5]
```diff
@@ -131,3 +131,12 @@ Network Security (IPS/IDS) Integration Example: [ProtectWise](https://xsoar.pan.
...
 Vulnerability Management Integration Example: [Tenable.io](https://xsoar.pan.dev/docs/reference/integrations/tenableio)
+
+## IAM (Identity and Access Management)
+
+**Top Use Cases:**
+- Create/update/delete users.
+- Block user/Force change password
+- Managing access to resources and applications.
+
+Implement the IAM generic CRUD commands as described [Here](../integrations/generic-iam-commands)
```
Comment (fvigo) [inline docs/concepts/use-cases.md:142]: We should remove this link until `generic-iam-commands` is complete

File: docs/integrations/context-standards-about.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+---
+id: context-standards-about
+title: About Context Standards
+---
+
+Cortex XSOAR organizes incident data in a tree of objects called the *Incident Context*. Any integration commands or scripts that run, will add data into the context at a predefined location. This also applies to commands that run within playbook execution.
```
Comment (fvigo) [inline docs/integrations/context-standards-about.md:6]: suggest to add a link here to `../concepts/concepts#context-data`

File: docs/integrations/context-standards-mandatory.md [hunk 7]
```diff
@@ -1,22 +1,12 @@
...
-When building new integrations the entry context should be returned according to this standard. 
-
-If there is no matching item in the standard, reach out to the architect team to check if your addition merits an update or change of the standard. 
-
-If the data is vendor/tool specific, it can be assigned its own context path.
-
-Below are examples of how each entity should be formed in the entry context.
-
 ## File
 The following is the format for a file.
```
Comment (fvigo) [inline docs/integrations/context-standards-mandatory.md:7]: Worth mentioning what we mean by `File` and the differences from `InfoFile` below.

File: docs/integrations/context-standards-mandatory.md [hunk 8]
```diff
@@ -669,131 +627,7 @@ The following is the format for an Event.
...
-- contextPath: Host.Processor
-  description: The processor of the host.
-  type: String
-- contextPath: Host.Processors
-  description: The number of processors that the host is using.
-  type: Number
-```
+``` 
 
 ## InfoFile
```
Comment (fvigo) [inline docs/integrations/context-standards-mandatory.md:632]: We should mention the differences from `File` above.

File: docs/integrations/context-standards-recommended.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+```
+
+**In YAML**
+```yaml
```
Comment (fvigo) [inline docs/integrations/context-standards-recommended.md:20]: We should add the YAML if possible

File: docs/integrations/context-standards-recommended.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+    "Type": "STRING, The type of event, for example: "ePO", "Protectwise", "DAM".",
+    "ID": "STRING, The unique identifier of the event.",
+    "Name": "STRING, The name of the event.",
+    "Sensor": "STRING, The sensor that indicated the event.",
+    "Rule": "STRING, The rule that triggered the event."
+}
+```
+
+**In YAML**
+```yaml
```
Comment (fvigo) [inline docs/integrations/context-standards-recommended.md:161]: We should add the YAML if possible

File: docs/integrations/context-standards-recommended.md [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+    "Name": "STRING, The name of the service.",
+    "BinPath": "STRING, The path of the /bin folder.",
+    "CommandLine": "STRING, The full command line (including arguments).",
+    "StartType": "STRING, How the service was started.",
+    "State": "STRING, The status of the service."
+}
+```
+
+**In YAML**
+```yaml
```
Comment (fvigo) [inline docs/integrations/context-standards-recommended.md:178]: We should add the YAML if possible

File: docs/integrations/context-standards-recommended.md [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+    "Start Time": "DATE, The timestamp of the process start time.",
+    "End Time": "DATE, The timestamp of the process end time.",
+    "Parent": "STRING, Parent process objects.",
+    "Sibling": "LIST, Sibling process objects.",
+    "Child": "LIST, Child process objects."
+}
+```
+
+**In YAML**
+```yaml
```
Comment (fvigo) [inline docs/integrations/context-standards-recommended.md:202]: We should add the YAML if possible

File: docs/integrations/generic-commands-reputation.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+title: Generic Reputation Commands
+---
+
+
+## Background and motivation
+
+XSOAR has an abudance of integrations with reputation providers for example VirusTotal, AlienVault OTX, MISP etc. 
+Every Integration that returns a reputation about an indicator must implement the generic repuation commands and calculate [DBot Score](../dbot)
+
+## Generic Reputation Commands needs to be supported
```
Comment (fvigo) [inline docs/integrations/generic-commands-reputation.md:12]:
```suggestion
## Generic Reputation Commands

If an integration returns reputation on specific indicator types, it must implement the commands according to these naming and functionality standards.
```

File: docs/integrations/generic-commands-reputation.md [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+
+### **file**
+
+**Description:** Runs reputation on a File.
```
Comment (fvigo) [inline docs/integrations/generic-commands-reputation.md:16]: We should mention the argument name as well, and that is mandatory and default, and that the command should return the `File` context standard with a link to the schema. Maybe add the YML of the command so they can easily copy and paste.

File: docs/integrations/generic-commands-reputation.md [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+
+### **ip**
+
+**Description:** Runs reputation on an IP.
```
Comment (fvigo) [inline docs/integrations/generic-commands-reputation.md:20]: same as above

File: docs/integrations/generic-commands-reputation.md [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+
+### **url**
+
+**Description:** Runs reputation on a URL.
```
Comment (fvigo) [inline docs/integrations/generic-commands-reputation.md:24]: same as above

File: docs/integrations/generic-commands-reputation.md [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+
+### **domain**
+
+**Description:** Runs reputation on a Domain.
```
Comment (fvigo) [inline docs/integrations/generic-commands-reputation.md:28]: same as above

File: docs/integrations/generic-commands-reputation.md [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+
+### **email**
+
+**Description:** Runs reputation on an Email.
```
Comment (fvigo) [inline docs/integrations/generic-commands-reputation.md:32]: same as above

File: docs/integrations/generic-commands-reputation.md [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+
+### **cve**
+
+**Description:** Runs reputation on a CVE.
```
Comment (fvigo) [inline docs/integrations/generic-commands-reputation.md:36]: same as above

**Review by fvigo — CHANGES_REQUESTED** [review]
Added a few comments, looks good overall.

File: docs/integrations/dbot.md [hunk 20]
```diff
@@ -35,6 +35,14 @@ Dbot uses an integer to represent the reputation of an indicator.
 | 2 | Suspicious |
 | 3 | Bad |
 
+## Unknown
+Unknown score can be interpeted in two ways: 
+
+1. The vendor returns an "Unknown" score for the indicator.
+2. The vendor returns nothing on the indicator.
```
Comment (anara123) [inline docs/integrations/dbot.md:42]: I think the descriptions here are a bit incorrect.
1. The indicator do not exists in the product/integration.
2. The indicators in the product have no score/vector (only data)

File: docs/integrations/context-standards-mandatory.md [hunk 21]
```diff
@@ -251,7 +209,7 @@ The following is the format for a ticket.
 
 ## Email Object
```
Comment (anara123) [inline docs/integrations/context-standards-mandatory.md:210]: You have duplicates in both files. Delete from here Event, Account and others

File: docs/integrations/generic-commands-about.md (hunk 3 above)
Comment (Arsenikr) [reply docs/integrations/generic-commands-about.md:11]: no we don't afaik

File: docs/integrations/generic-commands-reputation.md (hunk 14 above)
Comment (Arsenikr) [reply docs/integrations/generic-commands-reputation.md:16]: A YML here will be a copy paste of the context standards. maybe i can link to it instead.. WDYT?

File: docs/integrations/context-standards-mandatory.md [hunk 22]
```diff
@@ -4,7 +4,7 @@ title: Mandatory Context Standards
 ---
 
 ## File
-The following is the format for a file.
+The following is the format for a file. File here refers to the file indicator.
```
Comment (fvigo) [inline docs/integrations/context-standards-mandatory.md:7]:
```suggestion
The following is the format for a file. File here refers to the file indicator or a binary file that could potentially be malicious, and might be checked for reputation or sent to a sandbox. 
```

**Review by fvigo — CHANGES_REQUESTED** [review]
Just added some minor tweaks and I think it's good to go

File: docs/integrations/context-standards-mandatory.md [hunk 23]
```diff
@@ -1,22 +1,12 @@
 ---
-id: context-standards
-title: Context Standards
+id: context-standards-mandatory
```
Comment (glicht) [inline docs/integrations/context-standards-mandatory.md:2]: We need to add a redirect to the `_redirects` file

**Review by glicht — CHANGES_REQUESTED** [review]
@Arsenikr there are a few empty yaml examples as reported here: https://github.com/demisto/content-docs/issues/346 

Please add these too.

File: docs/concepts/use-cases.md (hunk 5 above)
Comment (Arsenikr) [reply docs/concepts/use-cases.md:142]: removed fully. will add when IAM design will be stable

File: docs/integrations/dbot.md (hunk 20 above)
Comment (Arsenikr) [reply docs/integrations/dbot.md:42]: about 1 i tend to agree it is better. 
about 2 - there is a case that the vendor actually returns "Unknown" for the indicator

so maybe: 
1. The vendor returns an "Unknown" score for the indicator.
2. The indicator do not exists in the product/integration.
3. The indicators in the product have no score/vector (only data)

WDYT

File: docs/integrations/context-standards-mandatory.md (hunk 23 above)
Comment (Arsenikr) [reply docs/integrations/context-standards-mandatory.md:2]: added one to context-standards-about

**Review by fvigo — APPROVED** [review]

**Comment by Arsenikr** [comment]
@kirbles19 please review

File: docs/concepts/use-cases.md [hunk 24]
```diff
@@ -131,3 +131,10 @@ Network Security (IPS/IDS) Integration Example: [ProtectWise](https://xsoar.pan.
 - Scan assets for a specific vulnerability.
 
 Vulnerability Management Integration Example: [Tenable.io](https://xsoar.pan.dev/docs/reference/integrations/tenableio)
+
+## IAM (Identity and Access Management)
```
Comment (kirbles19) [inline docs/concepts/use-cases.md:135]: Should this be: ILM (Identify Lifecycle Management)?

File: docs/concepts/use-cases.md [hunk 25]
```diff
@@ -131,3 +131,10 @@ Network Security (IPS/IDS) Integration Example: [ProtectWise](https://xsoar.pan.
...
+
+**Top Use Cases:**
+- Create/update/delete users.
+- Block user/Force change password
+- Managing access to resources and applications.
```
Comment (kirbles19) [inline docs/concepts/use-cases.md:140]: Manage access to resources and applications.

File: docs/concepts/use-cases.md [hunk 26]
```diff
@@ -131,3 +131,10 @@ Network Security (IPS/IDS) Integration Example: [ProtectWise](https://xsoar.pan.
 - Scan assets for a specific vulnerability.
 
 Vulnerability Management Integration Example: [Tenable.io](https://xsoar.pan.dev/docs/reference/integrations/tenableio)
+
+## IAM (Identity and Access Management)
+
+**Top Use Cases:**
+- Create/update/delete users.
+- Block user/Force change password
```
Comment (kirbles19) [inline docs/concepts/use-cases.md:139]: - Block a user
- For change of password

File: docs/integrations/context-standards-mandatory.md [hunk 27]
```diff
@@ -1,22 +1,12 @@
...
-
-If there is no matching item in the standard, reach out to the architect team to check if your addition merits an update or change of the standard. 
-
-If the data is vendor/tool specific, it can be assigned its own context path.
-
-Below are examples of how each entity should be formed in the entry context.
-
 ## File
-The following is the format for a file.
+The following is the format for a file. File here refers to the file indicator or a binary file that could potentially be malicious, and might be checked for reputation or sent to a sandbox.
```
Comment (kirbles19) [inline docs/integrations/context-standards-mandatory.md:7]: In this context, file refers to the file indicator or binary file that might be malicious and can be checked for reputation, or sent to a sandbox.

**Comment by kirbles19** [comment]
@Arsenikr Done.

File: docs/concepts/use-cases.md (hunk 24 above)
Comment (michalgold) [reply docs/concepts/use-cases.md:135]: The use case is "Identity and Access Management". We just gave a different more generic name to our pack.

**Comment by Arsenikr** [comment]
> @Arsenikr there are a few empty yaml examples as reported here: #346
> 
> Please add these too.

@glicht filled those. any additional changes? i want to close the PR

**Review by glicht — APPROVED** [review]
looks good

**Pull request merged**